spmd/tensor/ops/matrix_ops.py

```diff
@@ -16,50 +17,48 @@ def dist_addmm(
     input: Tensor,
     mat1: Tensor,
     mat2: Tensor,
-    # pyre-fixme[2]: Parameter must be annotated.
-    beta=1,
-    # pyre-fixme[2]: Parameter must be annotated.
-    alpha=1,
+    beta: float = 1,
+    alpha: float = 1,
 ) -> Tensor:
     # dist addmm:
+    # input:replicate   mat1: replicate,  mat2: replicate
     # input:shard(0)    mat1: shard(0),  mat2: replicate
     # input:shard(1)    mat1: replicate, mat2: shard(1)
-    # input:replicate   mat1: shard(0),  mat2: replicate
-    # input:replicate   mat1: replicate, mat2: shard(1)
-    # input:replicate   mat1: shard(0),  mat2: shard(1)
+    # input:partial   mat1: shard(1),  mat2: shard(0)
+
+    # only implemented combo with no comm for now
+    # TODO: implement all combinations
+
     local_input, local_mat1, local_mat2 = pytree.tree_map(
         unwrap_local_tensor, (input, mat1, mat2)
     )
     input_placement, mat1_placement, mat2_placement = pytree.tree_map(
         unwrap_single_placement, (input, mat1, mat2)
     )
     device_mesh = mat1.device_mesh
+    if (
```

Comment:
maybe we should add the deleted cases as well for completeness.

Reply:
Oh this should be part of Charlie's PR. Once this gets rebased, this part should be gone.